admin/load.php

```diff
@@ -200,39 +200,49 @@ function perflab_get_modules( $modules_root = null ) {
 	$module_files = array();
 	$modules_dir  = @opendir( $modules_root );
 
+	// Modules are organized as {focus}/{module-slug} in the modules folder.
 	if ( $modules_dir ) {
-		// phpcs:ignore WordPress.CodeAnalysis.AssignmentInCondition.FoundInWhileCondition
-		while ( ( $file = readdir( $modules_dir ) ) !== false ) {
-			if ( '.' === substr( $file, 0, 1 ) ) {
-				continue;
-			}
+		// Load modules from each focus area folder.
+		$focus_areas = perflab_get_focus_areas();
+		foreach ( array_keys( $focus_areas ) as $focus ) {
```

Comment:
Can you keep the previous approach here where we go through all directories instead? The original code here was agnostic of which modules and focuses exist, which makes it more flexible and also easier to test, because it's not coupled to the focus areas that exist - which we've even talked about potentially removing them to not be hard-coded, which would be another reason to keep the "just go over the directories" approach here.

I would suggest you reinstate the original `while` loop, but then you check for `is_dir` and consider every top-level directory within a focus directory; then open that dir and iterate through it to get the modules similar to today.

Reply:
sure, I actually had it that way, then switched to using the specific focus names instead. in [`9ecb11a` (#58)](https://github.com/WordPress/performance/pull/58/commits/9ecb11a1dd6224c271c90269c35194cbad7dcdb8) which I will revert

Reply:
reverted that commit, should be fixed